src/Microsoft.AspNetCore.Mvc.Razor/RazorPageBase.cs

```diff
@@ -83,10 +83,10 @@ public virtual TextWriter Output
         public IDictionary<string, RenderAsyncDelegate> PreviousSectionWriters { get; set; }
 
         /// <summary>
-        /// Gets or sets a <see cref="System.Diagnostics.DiagnosticSource"/> instance used to instrument the page execution.
+        /// Gets or sets a <see cref="System.Diagnostics.DiagnosticListener"/> instance used to instrument the page execution.
         /// </summary>
         [RazorInject]
-        public DiagnosticSource DiagnosticSource { get; set; }
+        public DiagnosticListener DiagnosticSource { get; set; }
```

Comment:
This is a breaking change, can't do it 😢 - You'd have to add another property or do the cast conditionally and cache it in a field

Reply:
🤔

Reply:
OK pranav rightly pointed out to me that this is for 3.0 sorry for the confusion. I'll re-look-over this with a more reasonable set of expectations